actgen/agents/directed_qnet.py

```diff
@@ -11,14 +11,15 @@ class DirectedQNet(MLP):
     """
     def __init__(self, n_inputs, n_outputs, n_hidden_layers, n_units_per_layer, device,
                  lr, agent_type, pin_other_q_values, optim='sgd'):
-        super().__init__(n_inputs, n_outputs, n_hidden_layers, n_units_per_layer, device)
+        super().__init__(n_inputs, n_outputs, n_hidden_layers, n_units_per_layer)
         assert optim in ['sgd', 'adam']
         self.optim = optim
         self.lr = lr
         assert agent_type in ['dqn', 'action_dqn']
         self.agent_type = agent_type
         self.pin_other_q_values = pin_other_q_values
         self.device = device
+        self.model = self.model.to(device)
```

Comment:
Is this line necessary? We should be able to call `to(device)` from the outside.

Reply:
ah ok, good point. 

After I put the model on the GPU using `to(device)` from the outside, I still should put all the necessary tensors in this `DirectedQnet` class on the GPU right?

Reply:
I'm not sure I understand the question, but I think this may answer it...

Suppose you have a model called `d = DirectedQnet([args]).to(device)`, and suppose `DirectedQnet.forward()` takes input argument `x`. Since `d` is on `device`, `x` needs to be on the same `device`. If `y = d(x)`, then `y` is also on the same `device` until you call `cpu()`.

Does that help?

Reply:
yea thanks!